ext/faiss/index.cpp

```diff
@@ -136,7 +165,17 @@ void init_index(Rice::Module& m) {
         auto distances = numo::SFloat({n, k});
         auto labels = numo::Int64({n, k});
 
-        self.search(n, objects.read_ptr(), k, distances.write_ptr(), labels.write_ptr());
+        index_search_args args = {
+          .index = &self,
+          .n = n,
+          .x = objects.read_ptr(),
+          .k = k,
+          .distances = distances.write_ptr(),
+          .labels = labels.write_ptr(),
+        };
+
+        rb_thread_call_without_gvl(index_search_func, &args, RUBY_UBF_IO, NULL);
```

Comment:
Should this be `RUBY_UBF_IO` or `NULL`?

Reply:
It's gotta be doing pure CPU. We populate an in-memory store with a shit ton of vectors and then all "search" should be doing is doing some vector distance + additional math over huge sets of numbers.

Reply:
I used `RUBY_UBF_IO` because Cursor told me to. I bow to our AI overlords.

Reply:
:<

```
Time taken: 4.434523000032641 seconds
Time taken to create threads: 0.00047199998516589403 seconds
Time taken with threads: 5.277729999972507 seconds
```